### PR TITLE
Responsive layout

### PR DESCRIPTION
**Pull request by denzel-morris**
- Site navigation collapses to hamburger menu (with search at the top of the list vs. bottom)
- Docs page tabs collapses to two breadcrumbs
- Docs page TOC is hidden on mobile
- Footer collapses with logo and copyright at the bottom instead of top
- Further adherence to Material Design principles with addition of ripple animations on buttons, links, inputs, etc.
- Completed TODO: homepage title and subtitle styling
- Added Google Analytics tracking with ID configured by _config.yml
- Few general refactorings to separate concerns into more appropriate files

### CODE REVIEW

File: _includes/layout/base/nav.html [hunk 1]
```diff
@@ -0,0 +1,32 @@
+<nav class="navbar">
+  <div class="container">
+    <div class="navbar-header">
+      <a class="navbar-brand clearfix" href="{{ site.baseurl }}/">
+        <img src="{{ site.baseurl }}/images/trans/5.png" alt="Forseti Security Logo" />
+      </a>
```
Comment (mirons-google) [inline _includes/layout/base/nav.html]: nit: I know I removed the title [here](https://github.com/denzel-morris/forseti-security/commit/c3cc860663c36e594322833627a34e8349f4bf5d#diff-70c103dc160c67e7a579206a58d26b94), but can you add it back, it's better there when you're not on the home page.

File: _sass/layouts/_homepage.scss [hunk 2]
```diff
@@ -11,6 +11,17 @@
     margin: 0 0 25px;
   }
 
+  .project-title {
+    color: #fff;
+    // font-size: 1.2em;
```
Comment (mirons-google) [inline _sass/layouts/_homepage.scss]: remove this if it isn't needed, thanks.

File: _sass/layouts/_homepage.scss [hunk 3]
```diff
@@ -11,6 +11,17 @@
...
+    font-size: 46px;
+    font-weight: bold;
+  }
+
+  .project-subtitle {
+    color: #ffc400;
```
Comment (mirons-google) [inline _sass/layouts/_homepage.scss]: Question, feel free to save this for another PR. Can we make the colors from the color palette variables that way we don't need to update them everywhere?

**Comment by denzel-morris** [comment]
Broke out all the colors into appropriate SASS variables, and reproduced all Material Design colors as variables (in _sass/base/_colors.scss) for future use.

**Comment by blueandgold** [comment]
Adding `cla: yes` label, as the PR author is covered by CLA.

**Pull request merged**